app/tests/tests_10/test_collections_endpoint.py

```diff
@@ -596,3 +598,85 @@ def test_unauthorized_collection_delete(self):
         self.assertStatusCode(
             401, response, msg="unauthorized and unimplemented collection delete was permitted."
         )
+
+
+class CollectionLinksEndpointTestCase(StacBaseTestCase):
+
+    def setUp(self):
+        self.client = Client()
+        client_login(self.client)
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        cls.factory = Factory()
+        cls.collection_data = cls.factory.create_collection_sample(db_create=True)
+        cls.collection = cast(Collection, cls.collection_data.model)
+        return super().setUpTestData()
+
+    def test_create_collection_link_with_simple_link(self):
+        data = self.collection_data.get_json('put')
+
+        path = f'/{STAC_BASE_V}/collections/{self.collection.name}'
+        response = self.client.put(path, data=data, content_type="application/json")
+
+        self.assertEqual(response.status_code, 200)
+
+        link = CollectionLink.objects.last()
+        assert link is not None
+        self.assertEqual(link.rel, data['links'][0]['rel'])
+        self.assertEqual(link.hreflang, None)
+
+    def test_create_collection_link_with_hreflang(self):
+        data = self.collection_data.get_json('put')
+        data['links'] = [{
+            'rel': 'more-info',
+            'href': 'http://www.meteoschweiz.ch/',
+            'title': 'A link to a german page',
+            'type': 'text/html',
+            'hreflang': "de"
+        }]
+
+        path = f'/{STAC_BASE_V}/collections/{self.collection.name}'
+        response = self.client.put(path, data=data, content_type="application/json")
+
+        self.assertEqual(response.status_code, 200)
+
+        link = CollectionLink.objects.last()
+        # for the type checker here. But it's of course also valuable for the test
+        # since pytest should be the future, maybe I can afford this?
+        assert link is not None
```

Comment:
Through the lens of typing, `link` is possibly None. I need to tell the type checker that it shouldn't be, otherwise it'll say that "hreflang" is not a known attribute of "None". `self.assertNotEqual` can't do that. Now elegantly, `assert` both satisfies the type checker as well as it provides an assertion in the context of testing (at least if pytest is used).

Now we probably don't want to mix the way we assert things in testing, that's why I'm a bit hesitant here. On the other hand, it annoys me that I can't use this elegant solution 😄

Of course, static type checking isn't a requirement in service-stac but I thought we might as well start with that...

Reply:
Ok thanks.

No strong opinion, either works for me. I would just suggest to adapt the comment to something like
```py
# Check for None with `assert` because `self.assertNotEqual` is not understood by the type checker.
```

Reply:
Agreed, thanks for the suggestion!!